src/Engine/Manager/Managers.cpp

```diff
@@ -39,4 +46,92 @@ void Hub::ShutDown() {
     delete particleManager;
     delete physicsManager;
     delete resourceManager;
+
+    shutdown = true;
+
+}
+
+void Hub::AddComponent(World* world, Component::SuperComponent* component, const std::type_info* componentType) {
+
+    std::string name = componentType->name();
+
+    if (name == "class Component::Animation *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::DirectionalLight *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Lens *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Listener *")
+        soundManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Material *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Mesh *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::ParticleEmitter *")
+        particleManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Physics *")
+        physicsManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::PointLight *")
+        renderManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::Script *")
+        scriptManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::SoundSource *")
+        soundManager->AddComponent(world, component, componentType);
+    else if (name == "class Component::SpotLight *")
+        renderManager->AddComponent(world, component, componentType);
+    else
+        Log() << name << " not assigned to a manager!" << "\n";
+
```

Comment:
This is a bad solution, but if we want to rework the entire system it's good enough as a temporary thing.